src/Engine/ProtoCore/Core.cs

```diff
@@ -762,6 +762,30 @@ public void GenerateExecutable()
             DSExecutable.CurrentDSFileName = CurrentDSFileName;           
         }
 
+        /// <summary>
+        /// Gets the size to be used for runtime tables of symbols, procedures and instruction streams.
+        /// Note: since blocks are stored consecutively but may have gaps due to procedures being deleted,
+        /// this is based on largest id rather than amount of blocks.
+        /// </summary>
+        internal int GetRuntimeTableSize()
+        {
+            // Due to the way this list is constructed, the largest id is the one of the last block.
+            var lastBlock = CompleteCodeBlockList.LastOrDefault();
+            // If there are no code blocks yet, then the required size for tables is 0.
+            // This happens when the first code block is being created and its id is being generated.
+            if (lastBlock == null)
+            {
+                return 0;
+            }
+
+            // If the last block has children, then its last child has the largest id.
+            if (lastBlock.children.Count > 0)
+            {
+                lastBlock = lastBlock.children.Last();
```

Comment:
I guess I had similar question here why we are checking children of last block but other than this, everything else looks fine to me..

Reply:
I logged a ticket for the undo/redo bug. Maybe we can revisit this after we have fixed that? I feel they may be related.

I could also place a comment here mentioning that.

Reply:
Thanks, would be good if you do that